### PR TITLE
add ubuntu canary tests

### PR DESCRIPTION
**Pull request by lanking520**
Ubuntu Canary test on 16.04, 18.04 with MXNet, PyTorch and TensorFlow

### CODE REVIEW

File: .github/workflows/canary-ubuntu.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - cron:  '0 10 * * *'
+  workflow_dispatch:
+
+jobs:
+  canary-test-ubuntu:
+    if: github.repository == 'aws-samples/djl-demo'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04]
```
Comment (stu1130) [inline .github/workflows/canary-ubuntu.yml:14]: do we want to test centos7 with old glibc here as well

File: canary/src/main/java/ai/djl/canary/CanaryTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (stu1130) [inline canary/src/main/java/ai/djl/canary/CanaryTest.java:2]: nit: 2020

File: canary/src/main/java/ai/djl/canary/CanaryTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        logger.info("----------Environment Variables----------");
+        System.getenv().forEach((k, v) -> logger.info(k + ": " + v));
+
+        logger.info("");
+        logger.info("----------Default Engine----------");
+        Engine engine = Engine.getInstance();
+        engine.debugEnvironment();
+
+        logger.info("");
+        logger.info("----------Device information----------");
```
Comment (stu1130) [inline canary/src/main/java/ai/djl/canary/CanaryTest.java:53]: does debugEnv have the info of device count already?

Comment (frankfliu) [reply canary/src/main/java/ai/djl/canary/CanaryTest.java:53]: No, it doesn't

File: .github/workflows/canary-ubuntu.yml (hunk 1 above)
Comment (frankfliu) [reply .github/workflows/canary-ubuntu.yml:14]: centos need use docker container

**Review by stu1130 — APPROVED** [review]

**Pull request merged**